larray_editor/editor.py

```diff
@@ -272,13 +263,14 @@ def setup_menu_bar(self):
         file_menu.addAction(create_action(self, _('Save Data &As'), triggered=self.save_data_as,
                                           statustip=_('Save all arrays as a session in a file')))
         recent_files_menu = file_menu.addMenu("Open &Recent Data")
-        for action in self.recent_file_actions:
+        actions = [QAction(self) for _ in range(RecentlyUsedList.MAX_RECENT_FILES)]
+        for action in actions:
             action.setVisible(False)
             action.triggered.connect(self.open_recent_file)
             recent_files_menu.addAction(action)
-        self.update_recent_file_actions()
+        self.recent_data_files.actions = actions
```

Comment:
I still think this is weird: you should either not handle actions in RecentlyUsedList at all, or initialize them **there**. An hybrid version where initialization happens in two different places is bound to create problems down the line.